### ISSUE TITLE
Unusable from extensions

### ISSUE THREAD

**Issue by avi-screenovate** [yonat/ContactsChangeNotifier#1]
While the change notification will not be sent to extensions, it would still be beneficial if the code could _build_ when targeting extensions.  This allows including it as a dependency with other code that is shared between an app and its extension(s).

The problematic code is accessing `UIApplication.shared` in ContactsChangeNotifier.swift:140.

**Comment by yonat** [comment]
Fixed in 1.0.2.

**Comment by avi-screenovate** [comment]
I appreciate the extremely quick turn-around.  I've had the opportunity to integrate and test your code, and it's top-notch!

**Comment by gavingt** [comment]
Unfortunately, this fix causes crashes in some circumstances. Reproduction steps:

1) Download project ZIP, unzip, run demo app
2) Put app in background
3) Edit a contact
4) Put app in foreground and observe the change
5) Put app in background
6) Edit another contact
7) Put app in foreground

The crash happens soon after #7:

```
/private/var/containers/Bundle/Application/C2FBE89C-EF3D-44A8-AB7F-2F0DA0E7F5EB/ContactsChangeNotifierDemo.app/ContactsChangeNotifierDemo
2022-10-31 19:05:28.174029-0700 ContactsChangeNotifierDemo[15843:681199] failure in dealloc of <BSServiceConnectionEndpointMonitor:0x2838e8fa0> (BSServiceConnectionEndpointMonitor.m:58) : must call invalidate before dealloc
Exception: EXC_BREAKPOINT (code=1, subcode=0x1e69a83e4)
```

```
-[BSServiceConnectionEndpointMonitor dealloc] 0x00000001e69a83e4
-[UIApplication .cxx_destruct] 0x00000001cb774d48
object_cxxDestructFromClass(objc_object*, objc_class*) 0x00000001e1143380
objc_destructInstance 0x00000001e11401b4
_objc_rootDealloc 0x00000001e1149a00
-[UIResponder dealloc] 0x00000001cab04220
-[UIApplication dealloc] 0x00000001cb75bcfc
AutoreleasePoolPage::releaseUntil(objc_object**) 0x00000001e1142b14
objc_autoreleasePoolPop 0x00000001e113ee54
os_activity_apply_f 0x00000001e32bf8cc
-[_CNContactsLogger postingNotification:] 0x00000001cdcdd818
__131-[CNChangesNotifierProxy receiveNotificationName:fromSender:saveIdentifier:userInfo:calledFromNotifierQueue:isFromExternalProcess:]_block_invoke_2 0x00000001cdcea7c8
-[CNChangesNotifierProxy receiveNotificationName:fromSender:saveIdentifier:userInfo:calledFromNotifierQueue:isFromExternalProcess:] 0x00000001cdd83b54
__58-[CNChangesNotifierProxy receiveExternalNotificationName:]_block_invoke 0x00000001cdce1320
__63-[_CNJumpToMainRunLoopScheduler performBlock:qualityOfService:]_block_invoke 0x00000001d51d83e8
__CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ 0x00000001c83fa5a4
__CFRunLoopDoBlocks 0x00000001c83fb500
__CFRunLoopRun 0x00000001c83933d8
CFRunLoopRunSpecific 0x00000001c83a6bc8
GSEventRunModal 0x00000001e4512374
-[UIApplication _run] 0x00000001cad1ab58
UIApplicationMain 0x00000001caa9c090
closure #1 (Swift.UnsafeMutablePointer<Swift.Optional<Swift.UnsafeMutablePointer<Swift.Int8>>>) -> Swift.Never in SwiftUI.KitRendererCommon(Swift.AnyObject.Type) -> Swift.Never 0x00000001d0200f24
SwiftUI.runApp<τ_0_0 where τ_0_0: SwiftUI.App>(τ_0_0) -> Swift.Never 0x00000001d012ee08
static SwiftUI.App.main() -> () 0x00000001d01100f4
static ContactsChangeNotifierDemo.ContactsChangeNotifierDemoApp.$main() -> () ContactsChangeNotifierDemoApp.swift:10
main ContactsChangeNotifierDemoApp.swift:0
start 0x00000001049b1da4
```


The only solution I could find is to revert back to using `UIApplication.shared`.

**Comment by yonat** [comment]
Thanks for the clear report. Fixed in 1.0.3.
